### ISSUE TITLE
MP config injection types

### ISSUE THREAD

**Issue by xstefank** [quarkusio/quarkus#1538]
According to [1] Quarkus app should be able to inject config properties like:

```java
@Inject
@ConfigProperty(name="myprj.some.dynamic.timeout", defaultValue="100")
private javax.inject.Provider<Long> timeout;
```

or

```java
@Inject @ConfigProperty(name="myPets") private List<String> myListPets;
```

The Provider injection fails with Arc as it cannot resolve injection [2] and second List injection fails because List converter is not used by default which I think it should be according to the spec [3]. It doesn't also work with Set or String[].

[1] https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configexamples.asciidoc#simple-dependency-injection-example
[2] https://github.com/xstefank/reproducers/tree/master/config-provided-prop
[3] https://github.com/xstefank/reproducers/tree/master/config-list-inject

**Comment by stuartwdouglas** [comment]
I think the Provider issue may already be fixed in the next release

**Comment by mkouba** [comment]
Well, the general problem with `javax.inject.Provider` injection was fixed in 0.12.0, see also https://github.com/quarkusio/quarkus/issues/1373 but I can see some NPE from within SmallRye when the `Provider` is used together with `@ConfigProperty`. I'll investigate it further.

**Comment by mkouba** [comment]
@xstefank The config-list-inject reproducer works fine with 0.12.0.
